src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -153,14 +153,9 @@ private static string GetCorrectCasedPath(string path)
 
                         break;
                     }
-                    else if (item.Contains('~'))
+                    else if (item.Contains('~') || item.IndexOfAny(Utils.Separators.StarOrQuestion) >= 0)
```

Comment:
We could create new static variable for all 3 chars.

Reply:
I'm not sure if this single instance merits it.  I don't foresee many instances where this subset of special characters would share a common categorization on a regular basis.  Your call.